File: test/arch/src/pritunl-1.3.662.15/dart/lib/models/server_host.dart
library server_host_mod;

import 'package:pritunl/model.dart' as mdl;

import 'package:angular/angular.dart' show Injectable;
import 'package:angular/angular.dart' as ng;

@Injectable()
class ServerHost extends mdl.Model {
  @mdl.Attribute('id')
  String id;

  @mdl.Attribute('name')
  String name;

  @mdl.Attribute('server')
  String server;

  @mdl.Attribute('status')
  String status;

  @mdl.Attribute('address')
  String address;

  ServerHost(ng.Http http) : super(http);

  String get url {
    var url = '/server/${this.server}/host';

    if (this.id != null) {
      url += '/${this.id}';
    }

    return url;
  }
}

File: test/arch/src/pritunl-1.3.662.15/www/models/serverHost.js
define([
  'jquery',
  'underscore',
  'backbone'
], function($, _, Backbone) {
  'use strict';
  var ServerHostModel = Backbone.Model.extend({
    defaults: {
      'id': null,
      'server': null,
      'name': null,
      'address': null
    },
    url: function() {
      var url = '/server/' + this.get('server') + '/host';

      if (this.get('id')) {
        url += '/' + this.get('id');
      }

      return url;
    }
  });

  return ServerHostModel;
});
